mitxgraders/helpers/mitmath/specify_domain.py

```diff
@@ -95,12 +107,33 @@ def make_shape_validator(shape):
     >>> validate_vec4('cat')
     Traceback (most recent call last):
     Invalid: received a str, expected a vector of length 4
+
+    Instead of specifying a tuple shape, you can speicfy 'square' to demand
+    square matrices of any dimension.
+    >>> validate_square = make_shape_validator('square')
+    >>> square2 = MathArray([[1, 2], [3, 4]])
+    >>> square3 = MathArray([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+    >>> validate_square(square2)
+    MathArray([[1, 2],
+           [3, 4]])
+    >>> validate_square(square3)
+    MathArray([[1, 2, 3],
+           [4, 5, 6],
+           [7, 8, 9]])
+    >>> validate_square(MathArray([1, 2, 3, 4]))
+    Traceback (most recent call last):
+    Invalid: received a vector of length 4, expected a square matrix
```

Comment:
This looks like a good place to make a `scalar_or_vector` input to implement `abs`.

Reply:
I decided just to hard-code the requirement in a separate absolute value function so I could add a custom message suggesting norm.